builtin/common/vector.lua

```diff
@@ -366,3 +368,22 @@ function vector.dir_to_rotation(forward, up)
 	end
 	return rot
 end
+
+if rawget(_G, "core") and core.set_read_vector and core.set_push_vector then
```

Comment:
in which cases does `core` not exist?

Reply:
When the vector unit tests are run without the other tests. https://github.com/minetest/minetest/blob/1e9640395468beb53f70303ef6b7aa72e395b7b4/builtin/common/tests/vector_spec.lua#L1-L2